src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -2707,6 +2707,12 @@ void Unit::_UpdateSpells(uint32 time)
         }
     }
 
+    for (auto nexctSpellCrit : m_spellCritMap)
+    {
+        if (!nexctSpellCrit.first.first)
```

Comment:
did you debug this code ? I'm quite sure a debug session will show that the NULL check has nothing to do with checking if a Unit* is still valid